product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,602 @@
+---
+title: Performing a Postgres major version rolling upgrade on a PGD cluster built with TPA
+navTitle: Upgrading Postgres major versions
+deepToC: true
+---
+
+## Upgrading Postgres major versions
+
+Upgrading a Postgres database's major version to access improved features, performance enhancements, and security updates is a common administration task. Doing the same for a Postgres Distributed (PGD) cluster deployed with Trusted Postgres Architect (TPA) is essentially the same process but performed as a rolling upgrade.
+
+The rolling upgrade process allows updating individual cluster nodes to a new major Postgres version while maintaining cluster availability and operational continuity. This approach minimizes downtime and ensures data integrity by allowing the rest of the cluster to remain operational as each node is upgraded sequentially. 
+
+The following overview of the general instructions and [worked example](#worked-example) help to provide a smooth and controlled upgrade process.
+
+### Prepare the upgrade
+
+To prepare for the upgrade, identify which subgroups and nodes you are trying to upgrade and note an initial upgrade order.
+
+To do this, connect to one of the nodes using SSH and run the `pgd show-nodes` command:
+
+```bash
+sudo -u postgres pgd show-nodes
+```
+
+The `pgd show-nodes` command shows you all the nodes in your PGD cluster and the subgroup to which each node belongs. Then we want to know which node is the write leader in each subgroup by running:
+
+```bash
+sudo -u postgres pgd show-groups
+```
+
+This command outputs a list of the different groups/subgroups running in your cluster and the write leader of each group. Write leaders need to be switched over to another node in their subgroup before they can be upgraded to maintain operational continuity, so you should upgrade them lastly when upgrading a subgroup of nodes to keep the number of planned switchovers to a minimum.
+
+Even though you have verified which node is the current write leader for planning purposes, note that the write leader of a subgroup could change to another node at any moment for operational reasons before you upgrade that node. Therefore, you still need to verify that a node is not the write leader just before upgrading that node.
+
+You now should have enough information to determine your upgrade order, one subgroup at a time, aiming to upgrade the identified write leader node last in each subgroup. 
+
+### Perform the upgrade on each node
+
+!!! Note
+To help prevent data loss, ensure that your databases and configuration files are backed up before starting the upgrade process.
+!!!
+
+Using the [preliminary order](#prepare-the-upgrade), perform the following steps on each node while connected via SSH:
+
+* **Confirm the current Postgres version**
```

Comment:
I am just following the format DJ asked me too so that it is in line with what he is doing elsewhere in the PGD docs: https://www.enterprisedb.com/docs/pgd/latest/admin-manual/installing/05-creating-cluster/ 🤷